crates/swc/tests/fixture/issue-2669/declare-namespace-var-default/input/index.ts

```diff
@@ -0,0 +1,4 @@
+declare namespace TEST {
+  export const VALUE = "value";
+}
+export default TEST;
```

Comment:
Possibly, it can be said tsc's behavior is strange, but tsc emits this:

```
export default TEST;
```

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEA7KBbEBnADlMCAqAogMp7wDeAUPPCAB6YD2MALvGA4uqwGoCCAMgFUC8ALzwARADcoEAK4gJAbgoBfCnUYt4oAGZQ5EVoRJKgA

Reply:
It's because `declare` means the name `Test` exists, but just not by current module.
I think the bahavior of `tsc` is correct.

I don't think this will not cause a problem in real-world code, but `tsc` is correct here.

Reply:
Sorry, I should've said tsc's behavior is "inconsitent".
I have another test related to this in `crates/swc/tests/fixture/issue-2669/declare-namespace-type-default`:

- input

```ts
declare namespace TEST {
  export type TYPE = "type";
}
export default TEST;
```

- swc

```js
export { };
```

- tsc

```js
export {};
```

From the look of this, your interpretation:

> It's because declare means the name Test exists, but just not by current module.

doesn't necessary explain tsc's behavior. I might be missing something though.

Reply:
Addressed in this commit https://github.com/swc-project/swc/pull/2742/commits/02b766c598ad8796fef57d085c3d566a80f989b9

Reply:
This new test failure discovered a bug. It is fixed in https://github.com/swc-project/swc/pull/2742/commits/f7b406302dae6fb9560a313083982f8b5ed97c82